src/sagemaker/workflow/pipeline.py

```diff
@@ -220,6 +226,20 @@ def to_request(self) -> RequestType:
             "Steps": list_to_request(self.steps),
         }
 
+    def to_dict(self) -> Dict[str, any]:
+        """Converts Pipeline object into dict
+
+        Returns:
+            A dictionary for the Pipeline object
+        """
+        return {
+            "Name": self.name,
+            "Parameters": repr(self.parameters),
+            "Steps": repr(self.steps),
+            "Pipeline Experiment Config": repr(self.pipeline_experiment_config),
+            "Sagemaker Session": self.sagemaker_session,
```

Comment:
PipelineExperimentConfig
Usually we don't put spaces in the key names. See example in `to_request()`

And we don't need the Sagemaker Session for table display purpose. Let's remove it